src/controller_model/contact.ts

```diff
@@ -49,37 +50,43 @@ export class ContactControllerModel implements threema.ControllerModel<threema.C
         this.$log = $log;
         this.$translate = $translate;
         this.$mdDialog = $mdDialog;
-        this.contact = contact;
+        if (contact === undefined) {
+            if (mode !== ControllerModelMode.NEW) {
+                throw new Error('ContactControllerModel: Contact may not be undefined for mode ' + mode);
+            }
+        } else {
+            this.contact = contact;
+        }
         this.webClientService = webClientService;
         this.mode = mode;
 
         switch (this.getMode()) {
             case ControllerModelMode.EDIT:
                 this.subject = $translate.instant('messenger.EDIT_RECEIVER');
-                this.firstName = this.contact.firstName;
-                this.lastName = this.contact.lastName;
+                this.firstName = this.contact!.firstName;
```

Comment:
Might make sense to do `const contact = this.contact!` for these purposes but I don't have a strong opinion here.

Reply:
Then TS complains about name shadowing of the `contact` variable. And calling the variable `editContact` is also ugly.

Let's keep the current approach for now :slightly_smiling_face: